### PR TITLE
Return an empty array if there are no features to lookup 

### PR DESCRIPTION
**Pull request by etorreborre**
in `FeatureLookups.sparseMapToArray`.

### CODE REVIEW

File: ivory-storage/src/main/scala/com/ambiata/ivory/storage/lookup/FeatureLookups.scala [hunk 1]
```diff
@@ -53,11 +53,14 @@ object FeatureLookups {
     new EntityFilterLookup(features.map(_.toString).asJava, entities.asJava)
 
   def sparseMapToArray[A : scala.reflect.ClassTag](map: List[(Int, A)], default: A): Array[A] = {
-    val max = map.map(_._1).max
-    val array = Array.fill(max + 1)(default)
-    map.foreach {
-      case (i, a) => array(i) = a
+    if (map.isEmpty) Array()
+    else {
+      val max = map.map(_._1).max
```
Comment (charleso) [inline ivory-storage/src/main/scala/com/ambiata/ivory/storage/lookup/FeatureLookups.scala]: There isn't a `maxOr` that we could use here is there? (Same for the `maxBy` in the test)

**Comment by charleso** [comment]
:pray:

File: ivory-storage/src/main/scala/com/ambiata/ivory/storage/lookup/FeatureLookups.scala (hunk 1 above)
Comment (etorreborre) [inline ivory-storage/src/main/scala/com/ambiata/ivory/storage/lookup/FeatureLookups.scala]: Not that I know of in the standard lib. Maybe in Scalaz.

Comment (markhibberd) [inline ivory-storage/src/main/scala/com/ambiata/ivory/storage/lookup/FeatureLookups.scala]: `maximum`/`maximumOf`/`maximumBy` if I remember correctly.

**Comment by etorreborre** [comment]
I used `maximum` and `maximumBy` from Scalaz.

**Pull request merged**